posts/opentelemetry-and-friends.md

```diff
@@ -0,0 +1,136 @@
+                otelPolicy
+            }
+        });
+
+        const fargateTaskDefinition = new aws_ecs.FargateTaskDefinition(this, 'TaskDef', {
+            memoryLimitMiB: 512,
+            cpu: 256,
+            taskRole: taskRole
+        });
+        let webcontainer = fargateTaskDefinition.addContainer("WebContainer", {
+            image: ContainerImage.fromDockerImageAsset(dockerImageServiceB),
+            logging: new aws_ecs.AwsLogDriver({ streamPrefix: 'ServiceB', mode: aws_ecs.AwsLogDriverMode.NON_BLOCKING })
+        });
+        webcontainer.addPortMappings({
+            containerPort: 8080,
+        });
+        fargateTaskDefinition.addContainer("Otel", {
+            image: aws_ecs.ContainerImage.fromRegistry('amazon/aws-otel-collector:latest'),
+            logging: new aws_ecs.AwsLogDriver({ streamPrefix: 'ServiceBOtel', mode: aws_ecs.AwsLogDriverMode.NON_BLOCKING }),
+            command: ['--config=/etc/ecs/ecs-default-config.yaml', '--set=service.telemetry.logs.level=DEBUG']
+        });
+
+        // // Create a load-balanced Fargate service and make it public
+        let serviceB = new aws_ecs_patterns.ApplicationLoadBalancedFargateService(this, "SampleOtelService", {
+            cluster: cluster,
+            cpu: 512,
+            desiredCount: 1,
+            taskDefinition: fargateTaskDefinition,
+            assignPublicIp: true, // this is purely to not require a NAT option.
+            memoryLimitMiB: 1024,
+            publicLoadBalancer: true,
+            healthCheckGracePeriod: cdk.Duration.seconds(10),
+        });
+        dockerImageServiceB.repository.grantPull(serviceB.taskDefinition.obtainExecutionRole());
+
+    }
+}
+```
+
+It's important to note, that X-ray currently requires a specific time based ID format, which Quarkus discuss in [this section](https://quarkus.io/guides/opentelemetry#id-generator). You effectively need to use the X-Ray IDGenerator, or nothing will appear in the X-ray console.
+
+The X-ray console looks like the following. This example demonstrates hitting a sample `/hello` endpoint, via the deployed load-balancer, hitting the Quarkus application, and then making a `HTTP GET` call to a Star Wars test endpoint: `https://www.swapi.tech/api/starships/3`.
+
+![x-ray-console](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2022/11/opentelemetry_xray.png)
+
+# Why/When Would I Use It?
+It's a great way of integrating metric and trace monitoring into new applications, or applications that don't have existing solutions in this space. I've excluded logging for the moment as the specification is still in a [draft status](https://opentelemetry.io/docs/reference/specification/status/) at the time of writing, but it is certainly an exciting development. This is a pretty fast moving space, which will mature and solidify over time, but it will be the way of integrating vendor-agnostic tracing/metric/logging functionality into your application moving forward.
```

Comment:
Add a conclusion and a call to action after this final section.

Reply:
I've added a conclusion just below this. Hopefully that makes it a little cleaner.